feat: Add optional maxDistance to raycast (#2012)
flame-engine/flame: doc/flame/collision_detection.md

works similarly but sends out multiple rays uniformly around the origin, or within an angle
centered at the origin.

> By default `raycast` and `raycastAll` scans for the nearest hit irrespective of how far it lies from

Reviewer: ```suggestion
By default, `raycast` and `raycastAll` scan for the nearest hit irrespective of how far it lies from
```

Author: Done👍🏻

> the ray origin. But some use cases might be interested in finding hits only within a certain range.

Reviewer: ```suggestion
the ray origin. But some use cases, it might be interesting to find hits only within a certain range.
```

Author: Made it "But **in** some use cases, it might be interesting to find hits ..."

> For such cases, an optional `maxDistance` can be provided for `raycast` and `raycastAll`.

Reviewer: (redundant)
```suggestion
For such cases, an optional `maxDistance` can be provided.
```

Author: Done 👍🏻
